Source/Core/Common/TraversalServer.cpp

```diff
@@ -160,28 +161,51 @@ static sockaddr_in6 MakeSinAddr(const TraversalInetAddress& addr)
   return result;
 }
 
-static void GetRandomBytes(void* output, size_t size)
+// Seeding RNGs correctly is tough. See
+// http://open-std.org/JTC1/SC22/WG21/docs/papers/2016/p0205r0.html
+// Note that std::random_device is supposed to be non-deterministic, but may not be on less common
+// systems. See https://stackoverflow.com/questions/18880654/
+class RandomDeviceSeedSequence
 {
-  static u8 bytes[8192];
-  static size_t bytesLeft = 0;
-  if (bytesLeft < size)
+public:
+  using result_type = std::uint32_t;
+  template <typename It>
+  void generate(It begin, const It& end)
   {
-    ssize_t rv = read(urandomFd, bytes, sizeof(bytes));
-    if (rv != sizeof(bytes))
+    for (; begin != end; ++begin)
     {
-      perror("read from /dev/urandom");
-      exit(1);
+      *begin = m_device();
     }
-    bytesLeft = sizeof(bytes);
   }
-  memcpy(output, bytes + (bytesLeft -= size), size);
+
+  // An object meeting the full SeedSequence requirements is supposed to expose the original values
+  // passed in to it (the seeds for the seed, if you will). This class does not actually do that,
+  // but defines the functions in case any code checks for them.
+  size_t size() const { return 0; }
+  template <typename It>
+  void param(It out) const
+  {
+  }
+
+private:
+  std::random_device m_device;
+};
+
+static void GetRandomBytes(unsigned char* output, size_t size)
+
+{
+  static RandomDeviceSeedSequence s_seed_seq;
+  static std::independent_bits_engine<std::default_random_engine, sizeof(unsigned char) * 8,
```

Comment:
`CHAR_BIT` in `<climits>` can be used instead of `sizeof(unsigned char) * 8`

Reply:
Done.